src/google/protobuf/metadata.h

```diff
@@ -59,8 +59,7 @@ namespace internal {
 class LIBPROTOBUF_EXPORT InternalMetadataWithArena {
```

Comment:
Arena related code is quite performance sensitive so I am concerned about making them out-of-line. For the immediate issue, could you make it only out-of-line when PROTOBUF_USE_DLLS is defined?